src/rust/en.nana/src/parser.rs

```diff
@@ -0,0 +1,149 @@
+use aidoku::{
+	prelude::*, error::Result, std::{String, ObjectRef}, std::Vec, std::html::Node,
+	Filter, FilterType, Manga, Page, MangaStatus, MangaContentRating, MangaViewer
+};
+
+const BASE_URL: &str = "https://nana.my.id";
+
+pub fn parse_search(html: Node, result: &mut Vec<Manga>) {
+	for page in html.select("#thumbs_container > .id1").array() {
+		let obj = match page.as_node() {
+			Ok(node) => node,
+			Err(_) => return,
+		};
+		
+		let a = obj.select(".id3 > a");
+		let id: String = a.attr("href")
+		.read()
+		.split('/')
+		.last()
+		.unwrap()
+		.into();
```

Comment:
this isn't formatted right, did you run `cargo clippy`?

Reply:
yeah i ran cargo clippy and sad it was okay